core/controllers/topic_viewer.py

```diff
@@ -179,3 +180,37 @@ def get(self, topic_name: str) -> None:
             )
         })
         self.render_json(self.values)
+
+
+class TopicNameHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """A data handler for checking if a topic with given name exists."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'topic_name': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'has_length_at_most',
+                    'max_value': constants.MAX_CHARS_IN_TOPIC_NAME
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
+
+    @acl_decorators.open_access
```

Comment:
Hi @GusGus17, Could you please explain why are we using `open_access` decorator here? Should we be giving access to everyone for this?

Reply:
Confirming open access is OK here (nothing sensitive is being returned).